lib/middleware/user-session/user-session.js

```diff
@@ -1,145 +1,161 @@
-    addUserIdTokenMethods(req, userId, userToken)
-
-    next()
-  })
-  return router
-}
-
-const external = {
-  init,
-  clearSession,
-  clearSessionAndRedirect,
-  createSessionCookie,
-  createSessionAndRedirect
-}
-
-module.exports = external
+require('@ministryofjustice/module-alias/register-module')(module)
+
+const express = require('express')
+const router = express.Router()
+const cookieParser = require('cookie-parser')
+
+const aes256 = require('aes256')
+
+const user = require('~/fb-runner-node/user/user')
+const { getInstanceProperty } = require('~/fb-runner-node/service-data/service-data')
+const { getUrl } = require('~/fb-runner-node/route/route')
+
+const {
+  PLATFORM_ENV,
+  ROUTES: {
+    ping: PING_URL,
+    keepAlive: KEEP_ALIVE_URL,
+    healthcheck: HEALTHCHECK_URL
+  }
+} = require('~/fb-runner-node/constants/constants')
+
+const defaultSessionTimeout = 60 // minutes
+
+let serviceSecret
+
+const fakeSession = {}
+
+let sessionName
+let secure
+let cookiePath
+let cookieMaxAge
+
+function genid (userId = user.createUserId(), userToken = user.createUserToken()) {
+  const userDigest = user.createUserDigest(userId, userToken, serviceSecret)
+  const userDigestBundle = {
+    userId,
+    userToken,
+    userDigest
+  }
+
+  return JSON.stringify(userDigestBundle)
+}
+
+function setSessionCookie (res, encryptedDigestBundle) {
+  res.cookie(sessionName, encryptedDigestBundle, {
+    httpOnly: true,
+    secure,
+    cookiePath,
+    maxAge: 60 * 1000 * cookieMaxAge
+  })
+}
+
+function createSessionCookie (res, userId, userToken) {
+  const digestBundle = genid(userId, userToken)
+  const encryptedDigestBundle = aes256.encrypt(serviceSecret, digestBundle)
+  setSessionCookie(res, encryptedDigestBundle)
+  return encryptedDigestBundle
+}
+
+function createSessionAndRedirect (res, redirect = 'page.start') {
+  createSessionCookie(res)
+  return res.redirect(getUrl(redirect))
+}
+
+function clearSessionCookie (res) {
+  res.clearCookie('sessionId')
+}
+
+function clearSessionAndRedirect (res, redirect = 'page.start') {
+  clearSessionCookie(res)
+  return res.redirect(getUrl(redirect))
+}
+
+function addUserIdTokenMethods (req, userId, userToken) {
+  req.getUserId = () => userId
+  req.getUserToken = () => userToken
+}
+
+function getSessionOptions (options = {}, cookieMaxAge = 60) {
+  return Object.assign({}, {
+    sessionName: 'sessionId',
+    secure: !!PLATFORM_ENV,
+    cookiePath: '/',
+    cookieMaxAge
+  }, options)
+}
+
+function init (options = {}) {
+  // set secure: true when proper environments
+  const sessionDuration = getInstanceProperty('service', 'sessionDuration', defaultSessionTimeout)
+  const sessionOptions = getSessionOptions(options, sessionDuration)
+
+  sessionName = sessionOptions.sessionName
+  secure = sessionOptions.secure
+  cookiePath = sessionOptions.cookiePath
+  cookieMaxAge = sessionOptions.cookieMaxAge
+  serviceSecret = sessionOptions.serviceSecret || '<NONE>'
+

```

Comment:
The previous version declared `router` then did `router.use()` then did another `router.use()` then returned `router`

The return from `router.use()` is `router` so these calls a) can be chained, b) can be chained on the return